### PR TITLE
Implemented pre_term_timeout including test

### PR DESCRIPTION
**Pull request by ejlangev**
Implements the same functionality but includes a test of the signal handling to make sure things are working properly.

See https://github.com/resque/resque/pull/1136 for a description of the use case for this code.

### CODE REVIEW

**Comment by ejlangev** [comment]
@yaauie Had totally forgotten about this until right now, will fix up the last case where the signal handling isn't quite right.  Are there other changes necessary before this is good to go?

File: lib/resque/tasks.rb [hunk 1]
```diff
@@ -19,6 +19,8 @@
         worker.very_verbose = ENV['VVERBOSE']
       end
       worker.term_timeout = ENV['RESQUE_TERM_TIMEOUT'] || 4.0
+      worker.pre_shutdown_timeout = ENV['RESQUE_PRE_SHUTDOWN_TIMEOUT'] || 0.0
```
Comment (hoffmanc) [inline lib/resque/tasks.rb]: Never mind... nothing is testing top-level environment variables.

Comment (ejlangev) [inline lib/resque/tasks.rb]: Cool, seems like the tests have errored out for this build.  I'm going to rebase and run them again.

**Comment by steveklabnik** [comment]
Hey there! It's been a while, sorry about that.

@hoffmanc and I are going to be working on Resque again, but this PR needs a rebase. If you get the chance, would you mind

1. reabasing, if you're up for it
2. if you're not up for it, that's cool, just let us know so we can investigate

Thanks! / sorry :frowning:

**Comment by ejlangev** [comment]
Yeah I'll give rebasing it a shot.

**Comment by steveklabnik** [comment]
@ejlangev any chance you've had the chance to do that?

**Comment by steveklabnik** [comment]
Okay, giving this a close until then. Just let me know!

**Comment by joakimk** [comment]
@ejlangev @steveklabnik we rebased it for you in #1514 since we needed it :)

**Comment by ejlangev** [comment]
@joakimk Thanks!

**Pull request closed without merging**